bot/exts/evergreen/wikipedia.py

```diff
@@ -1,114 +1,99 @@
+                 "/330px-Wikipedia-logo-v2.svg.png"
+WIKI_SNIPPET_REGEX = r'(<!--.*?-->|<[^>]*>)'
+WIKI_SEARCH_RESULT = (
+    "**[{name}]({url})**\n"
+    "{description}\n"
+)
 
 
-class WikipediaSearch(commands.Cog):
-    """Get info from wikipedia."""
+class WikipediaSearchErrors(Enum):
+    """Errors returned in wikipedia search function."""
 
-    def __init__(self, bot: commands.Bot):
-        self.bot = bot
-        self.http_session = bot.http_session
+    no_results = "Sorry, we could not find a wikipedia article using that search term."
+    api_issue = "Whoops, the Wikipedia API is having some issues right now. Try again later."
 
-    @staticmethod
-    def formatted_wiki_url(index: int, title: str) -> str:
-        """Formating wikipedia link with index and title."""
-        return f'`{index}` [{title}]({WIKIPEDIA_URL.format(title=title.replace(" ", "_"))})'
 
-    async def search_wikipedia(self, search_term: str) -> Optional[List[str]]:
-        """Search wikipedia and return the first 10 pages found."""
-        pages = []
-        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
-            try:
-                data = await response.json()
+class WikipediaSearch(commands.Cog):
+    """Get info from wikipedia."""
 
-                search_results = data["query"]["search"]
+    def __init__(self, bot: Bot):
+        self.bot = bot
 
-                # Ignore pages with "may refer to"
-                for search_result in search_results:
-                    log.info("trying to append titles")
-                    if "may refer to" not in search_result["snippet"]:
-                        pages.append(search_result["title"])
-            except client_exceptions.ContentTypeError:
-                pages = None
+    async def wiki_request(self, ctx: commands.Context, search: str) -> Optional[List[str]]:
+        """Search wikipedia search string and return formatted first 10 pages found."""
+        url = SEARCH_API.format(number_of_results=10, string=search)
+        async with self.bot.http_session.get(url=url) as resp:
+            if resp.status == 200:
+                raw_data = await resp.json()
+                number_of_results = raw_data['query']['searchinfo']['totalhits']
+
+                if number_of_results:
+                    results = raw_data['query']['search']
+                    lines = []
+
+                    for article in results:
+                        formatting = {
+                            'name': article['title'],
+                            'description': unescape(
+                                re.sub(
+                                    WIKI_SNIPPET_REGEX, '', article['snippet']
+                                )
+                            ),
+                            'url': f"https://en.wikipedia.org/?curid={article['pageid']}"
+                        }
+                        line = WIKI_SEARCH_RESULT.format(**formatting)
```

Comment:
Redundant variable `formatting`
```suggestion
                        line = WIKI_SEARCH_RESULT.format(
                            name=article['title'],
                            description=unescape(
                                re.sub(
                                    WIKI_SNIPPET_REGEX, '', article['snippet']
                                )
                            ),
                            url=f"https://en.wikipedia.org/?curid={article['pageid']}"
                        )
```